CRM/Utils/PDF/Document.php

```diff
@@ -95,4 +95,64 @@ public static function toTwip($value, $metric) {
     return \PhpOffice\PhpWord\Shared\Converter::pointToTwip($point);
   }
 
+  /**
+   * @param array $path  docx/odt file path
+   * @param string $type  File type
+   * @param bool $returnContent extract content of docx/odt file as a text, later used for token replacement
+   *
+   * @return string
+   *    Return filepath of created html copy of document OR extracted content of document
+   */
+  public static function docReader($path, $type, $returnContent = FALSE) {
+    if ($returnContent) {
+      return self::doc2Text($path, $type);
+    }
+
+    $newpath = str_replace('.' . $type, '.html', $path);
+    $fileType = ($type == 'docx') ? 'Word2007' : 'ODText';
+    if (!file_exists($newpath)) {
+      $phpWord = \PhpOffice\PhpWord\IOFactory::load($path, $fileType);
+      $phpWord->save($newpath, 'HTML');
+    }
+
+    $relPath = str_replace('/Users/monish/www/civicrm-master/', CRM_Utils_System::baseURL(), $newpath);
+    return $relPath;
+  }
+
+  /**
+   * Extract content of docx/odt file as text and later used for token replacement
+   * @param string $filePath Document file path
+   * @param string $type File type of document
+   *
+   * @return string
+   *   File content of document as text
+   */
+  public static function doc2Text($filePath, $type) {
```

Comment:
@colemanw I didn't found any method to extract the content as a text from document file so I can use it for token replacement, so here's my solution for now as per https://github.com/PHPOffice/PHPWord/issues/490